### PR TITLE
impr: align sdo and cmo docmosis generation methods (FPLA-1480)

### PR DESCRIPTION
**Pull request by swalker125**
**Before creating a pull request make sure that:**

- [ ] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerMidEventTest.java [hunk 1]
```diff
@@ -92,6 +92,8 @@ void shouldPostExpectedTemplateDataAndAddDocumentReferenceToOrderActionWhenGener
     private CaseDetails getCaseDetails() {
         CaseDetails caseDetails = populatedCaseDetails();
 
+        caseDetails.getData().put("dateOfIssue", LocalDate.now());
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerMidEventTest.java:95]: we are moving away from using LocalDate(Time), please use the Time wrapper instead, it can be autowired in

Comment (swalker125) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerMidEventTest.java:95]: in test? is not simpler for me to just add like this?

Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ActionCaseManagementOrderControllerMidEventTest.java:95]: In this case it probably doesn't actually matter

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Order.java [hunk 2]
```diff
@@ -7,17 +7,28 @@
...
     private final JudgeAndLegalAdvisor judgeAndLegalAdvisor;
     private final String dateOfIssue;
+
+    public boolean isSealed() {
+        return SEALED == orderStatus;
+    }
+
+    public boolean isDraft() {
+        return !isSealed();
+    }
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/Order.java:33]: JsonIgnores for this

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java [hunk 3]
```diff
@@ -1,18 +1,50 @@
...
 
 @Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class CommonCaseDataExtractionService {
+    private static final String DEFAULT = "BLANK - please complete";
+
+    private final HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration;
+    private final HearingVenueLookUpService hearingVenueLookUpService;
+
+    String getCourtName(String localAuthority) {
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java:45]: Do we want the methods in here to be package private? I know that with our package structure they can be but should they be public?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java [hunk 4]
```diff
@@ -41,12 +73,78 @@ String getHearingTime(HearingBooking hearingBooking) {
...
+
+                return DocmosisHearingBooking.builder()
+                    .hearingDate(getHearingDateIfHearingsOnSameDay(hearing).orElse(""))
+                    .hearingVenue(hearingVenueLookUpService.buildHearingVenue(hearingVenue))
+                    .preHearingAttendance(extractPrehearingAttendance(hearing))
+                    .hearingTime(getHearingTime(hearing))
+                    .hearingJudgeTitleAndName(judgeAndLegalAdvisor.getJudgeTitleAndName())
+                    .hearingLegalAdvisorName(judgeAndLegalAdvisor.getLegalAdvisorName())
+                    .build();
+            }
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java:124]: Thoughts on extracting this to a method to keep the lambda syntax

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/StandardDirectionOrderGenerationService.java [hunk 5]
```diff
@@ -161,12 +97,6 @@ private DocmosisRespondent buildRespondent(RespondentParty respondent) {
...
 
-    public String getApplicantName(Applicant applicant) {
-        return ofNullable(applicant.getParty())
-            .map(ApplicantParty::getOrganisationName)
-            .orElse("");
-    }
-
     private String formatTitle(int index, Direction direction, List<DirectionConfiguration> directionConfigurations) {
 
         // default values here cover edge case where direction title is not found in configuration. Reusable for CMO?
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/StandardDirectionOrderGenerationService.java:102]: For this I would say yes, CMO doesn't have a directionConfig so we could create an overloaded method that just takes in index and direction and then calls this method which also passes an empty list

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java (hunk 3 above)
Comment (swalker125) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java:45]: public would provide the same access to this method as package private, so I don't see why it needs to be declared

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java:53]: That is what I was thinking, was just wondering whether it was intentional or not

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java [hunk 6]
```diff
@@ -100,17 +106,18 @@ void shouldPopulateHiddenCCDFieldsInStandardDirectionOrderToPersistData() {
 
         CaseDetails caseDetails = CaseDetails.builder()
             .data(createCaseDataMap(directionWithShowHideValuesRemoved)
-                .put("dateOfIssue", dateNow())
+                .put("dateOfIssue", time.now().toLocalDate().toString())
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java:109]: `dateNow()` returns a local date from `time.now()`

Comment (lewisbirks) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java:109]: Also do we need the `.toString()`

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java [hunk 7]
```diff
@@ -59,15 +64,16 @@
...
 
     @MockBean
     private DocmosisDocumentGeneratorService documentGeneratorService;
 
     @MockBean
     private UploadDocumentService uploadDocumentService;
 
+    @Autowired
+    private Time time;
+
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerAboutToSubmitTest.java:76]: `Time` is autowired into `AbstractControllerTest`

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOControllerTest.java [hunk 8]
```diff
@@ -78,10 +79,10 @@
...
 class DraftCMOControllerTest extends AbstractControllerTest {
     private static final long CASE_ID = 1L;
-    private static final LocalDateTime TODAY = LocalDateTime.now();
-    private static final List<Element<HearingBooking>> HEARING_DETAILS = createHearingBookingsFromInitialDate(TODAY);
-    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM)
-        .localizedBy(Locale.UK);
+    private static final DateTimeFormatter FORMATTER = ofLocalizedDate(FormatStyle.MEDIUM).localizedBy(Locale.UK);
+
+    @Autowired
+    private Time time;
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOControllerTest.java:85]: Same

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerMidEventTest.java [hunk 9]
```diff
@@ -50,7 +55,8 @@
     @MockBean
     private UploadDocumentService uploadDocumentService;
 
-    private Document document = document();
+    @Autowired
+    private Time time;
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerMidEventTest.java:59]: Same

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerMidEventTest.java [hunk 10]
```diff
@@ -59,21 +65,32 @@
     @BeforeEach
     void setup() {
         DocmosisDocument docmosisDocument = new DocmosisDocument(SEALED_ORDER_FILE_NAME, PDF);
+        Document document = document();
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerMidEventTest.java:68]: Seems like an unnecessary change

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerMidEventTest.java [hunk 11]
```diff
@@ -59,21 +65,32 @@
...
+                .directionText("example")
+                .assignee(LOCAL_AUTHORITY)
+                .readOnly("No")
+                .build()));
+
         CaseDetails caseDetails = CaseDetails.builder()
-            .data(createCaseDataMap(buildTestDirections())
-                .put("dateOfIssue", dateNow())
+            .data(createCaseDataMap(directions)
+                .put("dateOfIssue", time.now().toLocalDate().toString())
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerMidEventTest.java:89]: Also mentioned above, seems like an unnecessary change

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/StandardDirectionOrderGenerationService.java [hunk 12]
```diff
@@ -1,208 +1,84 @@
...
-
     private final HearingBookingService hearingBookingService;
-    private final HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration;
     private final OrdersLookupService ordersLookupService;
-    private final HearingVenueLookUpService hearingVenueLookUpService;
-    private final CommonCaseDataExtractionService dataExtractionService;
+    private final CommonCaseDataExtractionService dataService;
 
-    public static final String DEFAULT = "";
+    public static final String DEFAULT = "BLANK - please complete";
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/StandardDirectionOrderGenerationService.java:33]: Didn't we just get rid of this?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Order.java [hunk 13]
```diff
@@ -1,23 +1,37 @@
...
     private final String dateOfIssue;
+
+    @JsonIgnore
+    public boolean isSealed() {
+        return SEALED == orderStatus;
+    }
+
+    @JsonIgnore
+    public boolean isDraft() {
+        return !isSealed();
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/Order.java:35]: I don't think in the current implementation allows for it but this would mean that something that an order with null is a draft. Is this intentional? Also just for consistencies sake here you are doing `draft = !sealed` whereas in the CMO you are doing `sealed = !draft` maybe it would be nicer if it was always one way round. If that is done there could be a default implementation in the interface which just does `!isDraft/Sealed()`

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java [hunk 14]
```diff
@@ -0,0 +1,329 @@
...
+
+        Direction direction = getDirection(title);
+
+        assertThat(service.baseDirection(direction, 1)).isEqualToComparingFieldByField(expectedDirection(title));
+    }
+
+    @Test
+    void shouldBuildBaseDirectionAndTrimDirectionTextStringWhenWhitespace() {
+        String title = "Example title";
+        String directionText = " " + "Example description" + " ";
```
[the remark this answers is not in the document]
Comment (swalker125) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseDataExtractionServiceTest.java:255]: thought doing it as above would make it super clear the spaces before and after are intended. I see no reason to change
